### PR TITLE
feat(core): implement layer priorities

### PR DESCRIPTION
**Pull request by peppsac**
Before this commit, commands issued by layers were stored per-host.
These commands queues are sorted by commands priority.

The problem caused by this design is that commands priority values
are not normalized and if 2 different types of layer (eg: color and pointcloud)
shared the same host, their commands ended up in the same queue.
Their ordering would be determined by comparing "apples and oranges": color
layer priorities are not comparable to pointcloud layer priorities.

To address this problem, this commit introduces 1 queue per layer
within the same host and adds an optional layer priority.

When the scheduler has an empty slot for a given host, it will first
select the layer (based on their relative priorities) and then pick
its top most command.

### CODE REVIEW

**Comment by peppsac** [comment]
This PR has one test failing, only when using node lts version.
@gchoqueux it's a test you added so let me know if you have any idea (especially since I don't understand where the expected values comes from: when running the `globe.html` example in a 800x600 window I don't get the same values at all).

**Comment by peppsac** [comment]
> @gchoqueux it's a test you added so let me know if you have any idea (especially since I don't understand where the expected values comes from: when running the globe.html example in a 800x600 window I don't get the same values at all).

Ping @gchoqueux ? (I'm considering removing this test since it doesn't make sense)

**Comment by gchoqueux** [comment]
test fixed in #742

File: test/scheduler_unit_test.js [hunk 1]
```diff
@@ -31,9 +31,10 @@ function cmd(layerId = 'foo', prio = 0) {
         layer: {
             id: layerId,
             protocol: 'test',
+            priority: prio,
         },
+        priority: 1,
```
Comment (zarov) [inline test/scheduler_unit_test.js]: Unecessary now

File: src/Core/Scheduler/Scheduler.js [hunk 2]
```diff
@@ -18,18 +18,58 @@ import CancelledCommandException from './CancelledCommandException';
...
-                return cmp;
-            },
-        }),
+        queue(command) {
+            const layer = command.layer;
+            let st = this.storages.get(layer.id);
+            if (!st) {
+                st = {
+                    q: new PriorityQueue({ comparator: queueOrdering }),
+                    priority: layer.priority || 1,
```
Comment (zarov) [inline src/Core/Scheduler/Scheduler.js]: Duplicate from below

**Review by zarov — COMMENTED** [review]
LGTM, I just don't understand the necessity to also add the dithering algorithm ? Won't the separation of each layer already doing the wanted job ?

**Comment by peppsac** [comment]
> Won't the separation of each layer already doing the wanted job ?

The dithering is there to answer this question:
 
I have N layer (so N queues) that want to execute 1 command; which command do I execute next ?

**Review by zarov — APPROVED** [review]
Ok, LGTM then

**Pull request merged**